homeassistant/components/media_player/__init__.py

```diff
@@ -1089,3 +1079,13 @@ def as_dict(self, *, parent: bool = True) -> dict:
             response["children"] = []
 
         return response
+
+    def calculate_children_class(self):
+        """Count the children media classes and calculate the correct class."""
+        self.children_media_class = MEDIA_CLASS_DIRECTORY
+        child_class_count = collections.Counter(
+            child.media_class for child in self.children
+        )
+        if len(child_class_count) == 1:
+            # there's only one child class
+            self.children_media_class = next(iter(child_class_count))
```

Comment:
With Counter you end up counting also values that you don't care about. Since we only care about the same, we can just compare it all against first child.

```suggestion
        if len(self.children) == 0:
            return
        
        proposed_class = self.children.media_class
        if all(child.media_class == proposed_class for child in self.children):
            self.children_media_class = proposed_class
```